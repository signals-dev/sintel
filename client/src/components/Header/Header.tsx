import React, { useState } from 'react';
import './header.scss';
import { connect } from 'react-redux';
import { Link, useLocation } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faChevronRight, faChevronLeft, faCaretDown } from '@fortawesome/free-solid-svg-icons';
import { getCurrentExperimentDetails } from 'src/model/selectors/experiment';
import { filterEventsByTagAction } from 'src/model/actions/datarun';
import { getSelectedExperiment } from '../../model/selectors/projects';
import { onUserLogoutAction } from '../../model/actions/users';
import { RootState } from '../../model/types';
import Dropdown from '../Common/Dropdown';

type StateProps = ReturnType<typeof mapState>;
type DispatchProps = ReturnType<typeof mapDispatch>;
type Props = StateProps & DispatchProps;

export const Header: React.FC<Props> = (props) => {
  const isSwitchVisible = props.selectedExperimentID ? 'active' : '';
  const { experimentDetails, filterByTags } = props;
  let location = useLocation();

  const currentView = location.pathname;
  const linkTo = currentView === '/' ? `/experiment/${props.selectedExperimentID}` : '/';

  const logoutUser = () => {
    props.userLogout();
    window.location.href = '/';
  };

  const [isInfoOpen, toggleInfo] = useState(false);
  const activeClass = isInfoOpen ? 'active' : '';

  const dropDownProps = {
    isMulti: true,
    closeMenuOnSelect: false,
    placeholder: 'Filter',
    onChange: filterByTags,
  };

  window.addEventListener('click', (evt: Event) => {
    if (currentView === '/') {
      return null;
    }
    const dropdown = document.querySelector('.exp-info');
    dropdown && !dropdown.contains(evt.target as Node) && toggleInfo(false);
    return null;
  });

  return (
    <header id="header" className="main-header">
      <div className="header-container">
        <Link to="/" className="logo">
          <b>MTV</b>
        </Link>
        <Link to={linkTo} className={`page-switch-btn ${isSwitchVisible}`}>
          <FontAwesomeIcon icon={currentView === '/' ? faChevronRight : faChevronLeft} />
        </Link>
        {currentView !== '/' && (
          <div className="header-left-wrapper">
            <div className="exp-info" onClick={() => toggleInfo(!isInfoOpen)}>
              <ul>
                <li className={activeClass}>
                  Details
                  <span>
                    <FontAwesomeIcon icon={faCaretDown} />
                  </span>
                  <ul>
                    <li>
                      <span>Pipeline:</span>
                      <span>{experimentDetails.pipeline}</span>
                    </li>
                    <li>
                      <span>Dataset:</span>
                      <span>{experimentDetails.dataset}</span>
                    </li>
                    <li>
                      <span>By:</span>
                      <span>{experimentDetails.created_by || 'Unknown'}</span>
                    </li>
                    <li>
                      <span>Project:</span>
                      <span>{experimentDetails.project}</span>
                    </li>
                  </ul>
                </li>
              </ul>
            </div>
            <div className="tag-wrapper">
              <Dropdown {...dropDownProps} />
            </div>
          </div>
        )}

        <div className="header-right-wrapper">
          <ul className="user-opts">
            <li>
              <button type="button" onClick={logoutUser} className="logout-button">
                Logout
              </button>
            </li>
          </ul>
          <div className="clear" />
        </div>
      </div>
    </header>
  );
};

const mapState = (state: RootState) => ({
  selectedExperimentID: getSelectedExperiment(state),
  experimentDetails: getCurrentExperimentDetails(state),
});

const mapDispatch = (dispatch: Function) => ({
  userLogout: () => dispatch(onUserLogoutAction()),
  filterByTags: (tags) => dispatch(filterEventsByTagAction(tags)),
});

export default connect<StateProps, DispatchProps, {}, RootState>(mapState, mapDispatch)(Header);
