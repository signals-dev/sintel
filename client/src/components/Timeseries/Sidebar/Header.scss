@import '../../../assets/sass/form';

.sidebar-heading {
    ul {
        float: none;
    }

    li {
        margin-right: 10px;
        &.signal-title {
            font-size: 20px;
        }
    }

    .switch-control {
        font-size: 14px;
    }
}
