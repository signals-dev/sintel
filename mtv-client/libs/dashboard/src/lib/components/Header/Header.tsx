import React from 'react';
import './header.scss';
import { connect } from 'react-redux';
import { Link, useHistory } from 'react-router-dom';
import { getSelectedExperiment } from '../../model/selectors/projects';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faChevronRight, faChevronLeft } from '@fortawesome/free-solid-svg-icons'
import { RootState } from '../../model/types';

const mapState = (state: RootState) => ({
  selectedExperimentID: getSelectedExperiment(state),
});

type StateProps = ReturnType<typeof mapState>;
type Props = StateProps;

const Header: React.FC<Props> = props => {
  const isSwitchVisible = props.selectedExperimentID ? 'active' : '';
  let history = useHistory();
  const currentView = history.location.pathname;
  const linkTo = currentView === '/' ? `/experiment/${props.selectedExperimentID}` : '/';
  return (
    <header id="header" className="main-header">
      <Link to="/" className="logo">
        <b>MTV</b>
      </Link>
      <Link to={linkTo} className={`page-switch-btn ${isSwitchVisible}`}>
        <FontAwesomeIcon icon={currentView === '/' ? faChevronRight : faChevronLeft} />
      </Link>
    </header>
  );
};

export default connect<StateProps, {}, {}, RootState>(mapState)(Header);
