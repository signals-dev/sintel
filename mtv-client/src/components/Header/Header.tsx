import React from 'react';
import './header.scss';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { getSelectedExperiment } from '../../model/selectors/projects';

export interface HeaderProps {
  isExperimentSelected?: any
}

const Header: React.FC<HeaderProps> = (props) => {
  const isSwitchActive = props.isExperimentSelected ? 'active left' : '';
  return (
    <header id="header" className="main-header">
      <a href="/" className="logo"><b>MTV</b></a>
      <a href="/" className={`page-switch-btn ${isSwitchActive}`}>Switch</a>
      <i className="fa fa-caret-right">Switch test</i>
    </header>);
};

Header.propTypes = {
  isExperimentSelected: PropTypes.string,
};

export default connect(state => ({
  isExperimentSelected: getSelectedExperiment(state),
}), null)(Header);
