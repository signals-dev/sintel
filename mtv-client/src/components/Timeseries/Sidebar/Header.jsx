import React from 'react';

const Header = ({ headerTitle, reviewPeriod, reviewRange, currentPeriod, isEditingEventRange }) => (
  <div className="period-control">
    <div className="sidebar-heading">{headerTitle}</div>
    <ul>
      <li>
        <button
          type="button"
          onClick={() => !isEditingEventRange && reviewPeriod('year')}
          className={reviewRange === 'year' || reviewRange === null ? 'active' : ''}
        >
          Year
        </button>
      </li>
      <li>
        <button
          type="button"
          onClick={() => !isEditingEventRange && reviewPeriod('month')}
          className={reviewRange === 'month' ? 'active' : ''}
          disabled={reviewRange === null}
        >
          Month
        </button>
      </li>
      <li>
        <button
          type="button"
          onClick={() => !isEditingEventRange && reviewPeriod('day')}
          className={reviewRange === 'day' ? 'active' : ''}
          disabled={reviewRange === null || currentPeriod.month === ''}
        >
          Day
        </button>
      </li>
    </ul>
    <div className="clear" />
  </div>
);

export default Header;
