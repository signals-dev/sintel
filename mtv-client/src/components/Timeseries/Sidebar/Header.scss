@import '../../../assets/sass/form';

.sidebar-heading{
    ul{
        float: none;
    }

    li{
        margin-right: 15px;
        vertical-align: middle;
        &.signal-title{
            font-size: 20px;
        }
    }

    .switch-control{
        font-size: 14px;
    }
}
