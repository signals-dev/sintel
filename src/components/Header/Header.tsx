import React from 'react';
import './header.scss';


const Header: React.FC = () => {
    return (
        <header id="header" className="main-header">
            <a href="/" className="logo"><b>MTV</b></a>
            <a href="/" className="page-switch-btn" data-bind="click: pageSwitch" />
        </header>
    );
}

export default Header;
