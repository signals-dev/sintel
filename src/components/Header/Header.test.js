import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';

describe('Should render header', () => {
    it('Renders <Header /> with enzyme', () => {
        const header = shallow(<Header />);
        expect(header).toMatchSnapshot();

        const logo = header.find('a.logo');
        expect(header).toMatchSnapshot();
        expect(logo).toHaveLength(1);
    });
});
