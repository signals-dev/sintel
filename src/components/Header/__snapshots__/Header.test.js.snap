// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render header Renders <Header /> with enzyme 1`] = `ShallowWrapper {}`;

exports[`Should render header Renders <Header /> with enzyme 2`] = `ShallowWrapper {}`;
